data-science-onramp/data-ingestion/setup-test.py

```diff
@@ -0,0 +1,210 @@
+import os
+import re
+
+import uuid
+
+from google.api_core.exceptions import GoogleAPICallError
+
+from google.cloud import dataproc_v1 as dataproc
+from google.cloud import storage
+from google.cloud.exceptions import NotFound
+
+import pytest
+
+waiting_cluster_callback = False
+
+# Set global variables
+project = os.environ['GCLOUD_PROJECT']
+region = "us-central1"
+zone = "us-central1-a"
+cluster_name = 'setup-test-{}'.format(str(uuid.uuid4()))
+bucket_name = 'setup-test-code-{}'.format(str(uuid.uuid4()))
+
+
+@pytest.fixture(autouse=True)
+def teardown():
+    yield
+
+    # Delete cluster
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': f'{region}-dataproc.googleapis.com:443'
+    })
+
+    try:
+        operation = cluster_client.delete_cluster(project, region,
+                                                  cluster_name)
+        operation.result()
+    except GoogleAPICallError:
+        pass
```

Comment:
I would not catch this. If it fails it should fail loudly.

Reply:
+1 - doing an exception of a pass is a no-no - even if you want to ignore an error, you should let the user know you're doing something.

Reply:
As an addition - I'm not sure what you're checking for here. Is this the failure that's thrown when it can't delete it because it's already been deleted? If that's the case, add a clarifying comment and instead of the pass, a print statement that says "Cluster already deleted"